### PR TITLE
Circuit breakers observability

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds basic observability for circuit breakers. Target state would be keeping breaker events in the database but for now it should be sufficient.
⚠️ This is a PR to feature branch.

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: saleor/graphql/app/types.py [hunk 1]
```diff
@@ -597,6 +609,16 @@ def resolve_brand(root: models.App, _info: ResolveInfo):
             return root
         return None
 
+    @staticmethod
+    def resolve_breaker_state(root: models.App, _info: ResolveInfo):
+        if not breaker_board:
+            return CircuitBreakerState.CLOSED
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:615]: The field is nullable so maybe when no breaker is used, it should return null to indicate that?

File: saleor/graphql/app/types.py [hunk 2]
```diff
@@ -524,6 +535,7 @@ class App(ModelObjectType[models.App]):
         required=True,
     )
     brand = graphene.Field(AppBrand, description="App's brand data.")
+    breaker_state = graphene.String(description="App sync webhooks operation state.")
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:538]: This looks more like an enum in the schema. I also think the description is rather vague, and we should explicitly say that this is the state of the app's circuit breaker. If we had an enum in the schema, we could also describe what each state value exactly means.

I know that there was a suggestion not to use the "circuit breaker" term in the schema, but in this case, how should we call it to make sure that it's understandable to the user? @patrys Do you have any suggestion?

File: saleor/graphql/app/utils.py [hunk 3]
```diff
@@ -40,3 +40,8 @@ def validate_app_is_not_removed(
                 )
             }
         )
+
+
+class CircuitBreakerState:
```
Comment (maarcingebala) [inline saleor/graphql/app/utils.py:45]: Enums or enum-like classes are usually stored in the module's main `__init__` file, so I suggest `saleor/app/__init__.py` for this one.

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/app/types.py (hunk 1 above)
Comment (tomaszszymanski129) [reply saleor/graphql/app/types.py:615]: I could do that, but I'm not sure we want to expose configuration details like this through the API?

Comment (maarcingebala) [reply saleor/graphql/app/types.py:615]: I'm just wondering if it's not confusing when no breaker is used and the field will return one of the breaker's states. But you can also look at this from this perspective - when there is no breaker, the state is always "closed". But then the field shouldn't be nullable.

File: saleor/graphql/app/types.py [hunk 4]
```diff
@@ -53,13 +54,28 @@
...
+    CircuitBreakerState,
+    CircuitBreakerStateEnum,
+)
 from .resolvers import (
     resolve_access_token_for_app,
     resolve_access_token_for_app_extension,
     resolve_app_extension_url,
 )
 
+# TODO: Remove the conditional when unit tests circular import is solved.
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:70]: We should be able to remove this conditional now.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**